.pre-commit-config.yaml

```
@@ -0,0 +1,134 @@
+    -   id: check-yaml
+    -   id: check-json
+    -   id: check-toml
+    -   id: check-executables-have-shebangs
+    -   id: check-shebang-scripts-are-executable
+
+      # git checks
+    -   id: check-merge-conflict
+    -   id: check-added-large-files
+    -   id: detect-private-key
+    -   id: check-case-conflict
+      # - id: detect-aws-credentials
+
+      # python checks
+    -   id: check-docstring-first
+    -   id: check-builtin-literals
+    -   id: debug-statements
+    -   id: requirements-txt-fixer
+    -   id: fix-byte-order-marker
+    -   id: name-tests-test
+        args: [--pytest-test-first]
+
+      # general quality checks
+    -   id: mixed-line-ending
+    -   id: trailing-whitespace
+        args: [--markdown-linebreak-ext=md]
+    -   id: check-executables-have-shebangs
+    -   id: end-of-file-fixer
+
+-   repo: https://github.com/AleksaC/hadolint-py
+    rev: v2.14.0
+    hooks:
+    -   id: hadolint
+
+-   repo: https://github.com/asottile/pyupgrade
+    rev: v3.21.0
+    hooks:
+    -   id: pyupgrade
+        args: [--py310-plus]
+
+-   repo: https://github.com/adamchainz/django-upgrade
+    rev: 1.29.0
+    hooks:
+    -   id: django-upgrade
+        args: [--target-version, "4.0"]
+
+-   repo: https://github.com/psf/black-pre-commit-mirror
+    rev: 25.9.0
+    hooks:
+    -   id: black
+        exclude: ^party/migrations/
+
+-   repo: https://github.com/adamchainz/blacken-docs
+    rev: 1.20.0
+    hooks:
+    -   id: blacken-docs
+        additional_dependencies:
+        -   black==25.9.0
+
+-   repo: https://github.com/pycqa/flake8
+    rev: 7.3.0
+    hooks:
+    -   id: flake8
+        exclude: "^party/migrations/"
+        args: [--max-line-length=119, --max-complexity=4,
+              --pytest-fixture-no-parentheses]
+        additional_dependencies: [flake8-bugbear, dlint, flake8-use-fstring,
+              flake8-builtins, pep8-naming, flake8-variables-names, flake8-fixme,
+          flake8-pytest-style, flake8-django,
+              flake8-comprehensions, flake8-tidy-imports]
+
+-   repo: https://github.com/pycqa/isort
+    rev: 7.0.0
+    hooks:
+    -   id: isort
+        name: isort (python)
+        args: ["--filter-files", "--force-single-line-imports"]
+
+#-   repo: https://github.com/PyCQA/bandit
+#    rev: 1.8.3
+#    hooks:
+#    -   id: bandit
+#        exclude: ^party/tests/
+
+-   repo: https://github.com/shellcheck-py/shellcheck-py
+    rev: v0.11.0.1
+    hooks:
+    -   id: shellcheck
+
+-   repo: https://github.com/astral-sh/ruff-pre-commit
+    # Ruff version.
+    rev: v0.14.1
+    hooks:
+      # Run the linter.
+    -   id: ruff
+        args: [--fix]
+        exclude: ^party/migrations/
+      # Run the formatter.
+    -   id: ruff-format
+        exclude: ^party/migrations/
+
+-   repo: https://github.com/executablebooks/mdformat
+    rev: 0.7.22
+    hooks:
+    -   id: mdformat
+        additional_dependencies:
+        -   mdformat-mkdocs[recommended]>=4.1.2
```

The latest version is this. The additional_dependencies don't get automatically updated.
```suggestion
        -   mdformat-mkdocs[recommended]>=4.4.1
```